aws/cf_template.go

```diff
@@ -149,7 +155,7 @@ func generateTemplate(spec *stackSpec) (string, error) {
 			},
 			LoadBalancerArn: cloudformation.Ref("LB").String(),
 			Port:            cloudformation.Integer(80),
-			Protocol:        cloudformation.String(protocol),
+			Protocol:        cloudformation.String(httpProtocol),
```

Comment:
It will make NLB have the `HTTP` protocol and it's not supported.

We should have it tested too. NLB's `HTTPListener` should be checked to be using `TCP`, not `HTTP`.

Reply:
I think the following might be sufficient:
```diff
diff --git aws/cf_template_test.go aws/cf_template_test.go
index 16860e9..fc60f1c 100644
--- aws/cf_template_test.go
+++ aws/cf_template_test.go
@@ -161,6 +161,7 @@ func TestGenerateTemplate(t *testing.T) {
                        validate: func(t *testing.T, template *cloudformation.Template) {
                                require.NotNil(t, template.Resources["HTTPListener"])
                                listener := template.Resources["HTTPListener"].Properties.(*cloudformation.ElasticLoadBalancingV2Listener)
+                               require.Equal(t, listener.Protocol, cloudformation.String("TCP"))
                                require.Len(t, *listener.DefaultActions, 1)
                                redirectConfig := []cloudformation.ElasticLoadBalancingV2ListenerAction(*listener.DefaultActions)[0].RedirectConfig
                                require.Nil(t, redirectConfig)

```

Reply:
Good point @jonathanbeber 

Another thing is that you can't change the protocol of the health check on an NLB, so we actually need to handle that case when deciding which LBs to create, or at least document that changing the flag when you already have existing NLBs is not supported

Reply:
Thanks, good catch. Adjusted both.

Reply:
> Another thing is that you can't change the protocol of the health check on an NLB, so we actually need to handle that case when deciding which LBs to create, or at least document that changing the flag when you already have existing NLBs is not supported

I believe we can make it simpler by just having this affect ALBs. That was essentially my intention in the first place. I'll push another change up to reflect that, as well as specific tests to demonstrate NLBs are not impacted.